### DISCUSSION TITLE
SIGSEV/SIGABRT error with METIS partitioning

### DISCUSSION THREAD

**Discussion by LuisaWerner** [pyg-team/pytorch_geometric#5297] [Q&A] [answered]
Hi, 

I have problems with the METIS partitioning in the Cluster Loader. 
In my code, I want to apply the ClusterLoader to the ```ogbn-arxiv``` data set. 
I calculate the Clusters as follows:

```
cluster_data = ClusterData(data, num_parts=num_partitions, recursive=False)

train_loader = ClusterLoader(
            cluster_data=cluster_data,
            batch_size=round(args.batch_size / args.cluster_partition_size) + 1,
            shuffle=True,
            num_workers=args.num_workers)
```

When I start the training, the message "Computing METIS partitioning... " appears and then I get errors that are not deterministic and seem to depend on the sizes of the clusters. I received the following errors for example: 

```
Computing METIS partitioning...
munmap_chunk(): invalid pointer
munmap_chunk(): invalid pointer
munmap_chunk(): invalid pointer

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

```
Computing METIS partitioning...

Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```

```
Computing METIS partitioning...
free(): invalid size
free(): invalid size
corrupted size vs. prev_size

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

The errors seem only to occur together with ogbn-arxiv. I didn't experience them with ogbn-products. Also in the example code given for cluster_gcn on ogbn-products, these errors occur when the data set is replaced by arxiv. 

Does anyone know what causes this error and where it comes from? I would also like to apply ClusterLoader to ogbn-arxiv. 

Thanks for your help :-)

**Comment by rusty1s** [comment, marked as answer]
Can you try to convert the graph to an undirected graph first? I assume `METIS` does only work properly on undirected graphs:
```python
data = T.ToUndirected()(data)
```

**Reply by LuisaWerner** [reply]
Thanks for your fast reply! This solved the problem. It would be great to add this hint to the documentation. :-)

**Reply by rusty1s** [reply]
Indeed. Added a hint to the doc :)
